internal/provider/connection_resource.go

```diff
@@ -847,6 +892,18 @@ func (r *connectionResource) ValidateConfig(
 				validateRequiredString(plan.AWSAssumeRole.AccountRoleName, "aws_assume_role.account_role_name", "S3", resp)
 			}
 		}
+	case "postgresql":
+		validateRequiredString(plan.Host, "host", "PostgreSQL", resp)
+		validateRequiredInt(plan.Port, "port", "PostgreSQL", resp)
+		validateRequiredString(plan.UserName, "user_name", "PostgreSQL", resp)
+		validateRequiredString(plan.Password, "password", "PostgreSQL", resp)
+		validateRequiredString(plan.SSLMode, "ssl_mode", "PostgreSQL", resp)
```

Comment:
If SSL is disabled, there is no need to make `ssl_mode` required.

See below for information.
```suggestion
		if plan.SSL != nil {
			validateRequiredString(plan.SSLMode, "ssl_mode", "PostgreSQL", resp)
		}
```

Reply:
it looks good!
however, `ssl_mode` is not required for the other connector (like mysql),
so please continue to set `ssl_mode` with `Optional: true`, but validate in `ValidateConfig`.